### PR TITLE
added Queue implementation with linked list in C

### PR DESCRIPTION
**Pull request by hushee69**
Please read [how to contribute](CONTRIBUTING.md) first!
# Description
Submission description here

Checklist
* [x] I've checked the issue list __BEFORE__ deciding what to submit
* [x] I've read the contribution guidelines
* [x] I've edited the README.md link to my code

## Don't forget to visit our [website](http://www.tesseractcoding.tech)!

Feel free to add this badge after your submission is accepted!

```
<a href=https://github.com/TesseractCoding/NeoAlgo>
   <img src=https://img.shields.io/badge/NeoAlgo-Contributor-brightgreen>
</a>
```

## Problem being solved: Issue number 316

Added algorithm in `ds` for Queue in C using linked lists.

### CODE REVIEW

**Comment by anushkrishnav** [comment]
Please edit the README.md file to add a link to you file . Thank you :)

File: C/ds/QueueImplementationWithList.c [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        printf(" %d ", cur->value);
+        cur = cur->next;
+    }
+    printf("]\n");
+
+    return;
+}
+
+int main()
+{
```
Comment (atarax665) [inline C/ds/QueueImplementationWithList.c:81]: Please remove the extra empty lines.

File: C/ds/QueueImplementationWithList.c [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    }
+    else
+    {
+        // go until end of queue
+        cur = cur->head;
+        while( cur->next )
+        {
+            cur = cur->next;
+        }
+
```
Comment (atarax665) [inline C/ds/QueueImplementationWithList.c]: Please remove the extra empty lines.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by atarax665** [comment]
Please fill the PR template. Mention the issue that your PR is fixing.

**Comment by hushee69** [comment]
> 
> 
> Please fill the PR template. Mention the issue that your PR is fixing.

done

**Comment by iamrajiv** [comment]
@all-contributors please add @hushee69 for code

**Pull request merged**